src/lib/xmlparse.rs

```diff
@@ -4165,14 +4120,12 @@ impl XML_ParserStruct {
                     }
                     if self.m_handlers.hasCharacterData() {
                         if MUST_CONVERT!(enc, buf.as_ptr()) {
-                            let mut dataPtr = ExpatBufRefMut::new(
-                                self.m_dataBuf as *mut ICHAR,
-                                self.m_dataBufEnd as *mut ICHAR,
-                            );
+                            let dataStart = self.m_dataBuf.as_ptr();
+                            let mut dataPtr = (&mut self.m_dataBuf[..]).into();
                             XmlConvert!(enc, &mut buf, &mut dataPtr);
                             self.m_handlers.characterData(
                                 &ExpatBufRef::new(
```

Comment:
If I'm reading this correctly, `dataPtr` is a `ExpatBufRef`, so couldn't you call `with_start` here instead of building a new object?

Reply:
It would actually be `with_end(dataPtr.as_ptr())`, but yeah.